### DISCUSSION TITLE
Support for Content-Type: application/graphql

### DISCUSSION THREAD

**Discussion by dominikbrandon** [Netflix/dgs-framework#34] [Ideas]
Do you plan on adding support for `Content-Type: application/graphql` header, like `graphql-java-spring` [does](https://github.com/graphql-java/graphql-java-spring/blob/master/graphql-java-spring-webmvc/src/main/java/graphql/spring/web/servlet/components/GraphQLController.java)? I find it more convenient in use with manually built requests. With JSON one needs to supply whole query in a single string, which can't be broken into lines. It's not very readable even for simple queries. What do you think?

Cheers!

**Comment by srinivasankavitha** [comment]
For manually building the requests - have you checked out out the GraphQL client functionality that can be used with codegen? It provides a query builder API that you can use to construct your query and serialize into a string. The Type-safe Query API section described how to set that up.
https://netflix.github.io/dgs/advanced/java-client/

**Comment by paulbakker** [comment]
Is the difference that the query, operation name, etc., are all provided a separate request parameters? Is that for GET requests only? I’m not familiar with application/GraphQL so I’m trying to understand what it is and why it’s useful.

Also, are there commonly used clients that support this?

**Comment by dominikbrandon** [comment]
Thanks guys for your input. My description could have misled you - what I meant by manually built requests was this kind of manual requests:
```
POST http://localhost:8080/graphql
Content-Type: application/json

{
  "query": "{ allBooks { title, id } }"
}
```
I use this kind of raw HTTP requests with IntelliJ, which has nice features for it (in case you've missed it: https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html).

I don't know if there are any clients supporting this, it's still new to me. My point is: why write requests [like this](https://github.com/dominikbrandon/graphql-sandbox/blob/main/etc/requests/graphql-json.http) when you can achieve the same thing [that way](https://github.com/dominikbrandon/graphql-sandbox/blob/main/etc/requests/graphql.http).

**Reply by paulbakker** [reply]
I wasn't familiar with the support for application/graphql in Intellij, but that looks great. I think it would be good to support this in DGS as well.

**Comment by paulbakker** [comment]
Added an issue for this new feature: https://github.com/Netflix/dgs-framework/issues/43

**Reply by dominikbrandon** [reply]
Great, thanks!

**Comment by paulbakker** [comment]
This is now supported! See PR https://github.com/Netflix/dgs-framework/pull/166 for context

**Reply by dominikbrandon** [reply]
I've checked this out and it works like a charm, thank you!
